### DISCUSSION TITLE
How can I construct a query WHERE clause that always uses `=`, even if the RHS value is null?

### DISCUSSION THREAD

**Discussion by andremarianiello** [sequelize/sequelize#15922] [Help & Questions]
Hello! I am having some issues with WHERE clauses. For the record, I am using v6.29.3, but I believe this same behavior occurs all the way up to the latest version as well. To illustrate I am going to use simplified versions of the real queries I am using.

When I write a query to find some rows, it looks like this:
```
Post.findAll({
  where: {
    authorId: myAuthorId
  }
});
// SELECT * FROM post WHERE authorId = 2;
```
This is exactly as expected when `myAuthorId` is 2. However, the story is a little different when `myAuthorId` is `null`.

```
Post.findAll({
  where: {
    authorId: myAuthorId
  }
});
// SELECT * FROM post WHERE authorId IS NULL;
```

The `=` becomes `IS`. This was a little unexpected when I first saw it, but ultimately is makes sense. 99% of the time when someone passes `null` as the value, they really want `IS NULL` behavior, i.e. to match rows where the `authorId` is null. However, I am in the 1% of cases where I want the `= NULL` behavior, i.e. match no rows. (If this distinction is confusing, look at https://dev.mysql.com/doc/refman/8.0/en/working-with-null.html). So I think, no problem, I'll explicitly request `Op.eq` so sequelize knows I want `= NULL` behavior.

```
Post.findAll({
  where: {
    authorId: {
      [Op.eq]: myAuthorId,
    }
  }
});
// SELECT * FROM post WHERE authorId IS NULL;
```

As you can see, even though I explicitly request `=` with `Op.eq`, sequelize _still_ insists on using `IS`. At this point I dug through the code and found in both v6.29.3 and on main code explicitly written to handle this. Since this behavior is obviously intentional, I decided not to file a bug and instead start a discussion, because I am stuck. This background leads me to my ultimate question:

tl;dr: How do I write a WHERE clause to generate a query with `= NULL` in sequelize v6 without writing the whole query as a raw query?

**Comment by ephys** [comment]
Late response, you likely already figured it out, but here is how you can do this:

```ts
Post.findAll({
  where: {
    authorId: {
      [Op.eq]: myAuthorId === null ? literal('NULL') : myAuthorId,
    }
  }
});
```
